### PR TITLE
Add help for the ANSI Support Preferences page

### PR DESCRIPTION
**Pull request by mihnita**
https://github.com/eclipse-platform/eclipse.platform.debug/issues/47

### CODE REVIEW

**Comment by mihnita** [comment]
Corresponding PR adding the help context id to the Preferences dialog in Eclipse:
https://github.com/eclipse-platform/eclipse.platform.debug/pull/62

**Comment by mihnita** [comment]
@SarikaSinha : Thank you!

File: bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    <link rel="stylesheet" href="../../../book.css" charset="ISO-8859-1" type="text/css" />
+    <title>
+      Ansi Support Preferences
+    </title>
+    <script language="JavaScript" src=
+      "PLUGINS_ROOT/org.eclipse.help/livehelp.js" type="text/javascript">
+    </script>
+  </head>
+  <body>
+    <h1>Ansi Support Preferences</h1>
```
Comment (iloveeclipse) [inline bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm:18]: ANSI

File: bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+    <p>The following preferences can be set using the <a class="command-link" href=
+        'javascript:executeCommand("org.eclipse.ui.window.preferences(preferencePageId=org.eclipse.ui.internal.console.ansi.preferences.AnsiConsolePreferencePage)")'>
+      <img src="PLUGINS_ROOT/org.eclipse.help/command_link.svg" alt=
+        "Opens the Ansi Support preference page" /><b>Run/Debug &gt; Console &gt; Ansi Support</b></a> preference page.</p>
```
Comment (iloveeclipse) [inline bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm:23]: ANSI

File: bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+          </th>
+          <th rowspan="1" colspan="1">
+            <p>Default</p>
+          </th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <td rowspan="1" colspan="1" valign="top" align="left">
+            <p>Plugin enabled</p>
```
Comment (iloveeclipse) [inline bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm:55]: Please update preference label according to the current page state

**Review by iloveeclipse — DISMISSED** [review]
I haven't tried in the IDE, just quickly checked in browser. Please double check preferences labels for Ansi -> ANSI and other recent changes.

File: bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm (hunk 1 above)
Comment (mihnita) [reply bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm]: Done.

File: bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm (hunk 2 above)
Comment (mihnita) [reply bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm]: Done.

File: bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm (hunk 3 above)
Comment (mihnita) [reply bundles/org.eclipse.jdt.doc.user/reference/preferences/run-debug/ref-console_ansi.htm]: Done.

**Comment by mihnita** [comment]
I've managed to test things, (and in the process I've discovered that `<br>` had to be `<br />`, xml style)
But I checked that the options as described in the help now match 100% the real UI.

The only area where I was more "lax" was in `contexts_Debugger.xml`, where things seem to me more descriptive.
For example _"On this page, you can set the preferences for the debug console."_, no _"Debug Console"_ and _"Preferences"._
So I used _"On this page, you can set the ANSI support preferences for the console."_

**Comment by mihnita** [comment]
Managed to test that it works.

See screenshots with everything in place:
* https://mihai-nita.net/tmp/AnsiHelpInvoked.png
* https://mihai-nita.net/tmp/AnsiHelpPrefs.png
* https://mihai-nita.net/tmp/AnsiHelpTOC.png

**Comment by merks** [comment]
@mihnita Did you know that you can copy an image to the clipboard and paste it into the browser so that the image appears inline?  It's one of my favorite features of github issues.  On Windows I use Snip & Sketch to copy things like this:

image

Image editors also allow you to copy to the clipboard for this purpose...

**Comment by laeubi** [comment]
One can also embeds external images if one likes, I adjusted the comment so it shows the screenshots directly.

**Comment by mihnita** [comment]
Thank you, I didn't know.
I looked for some kind of "insert image" button, but I couldn't find any.
Well, now I know :-)

**Comment by laeubi** [comment]
> I looked for some kind of "insert image" button, but I couldn't find any.

grafik

**Comment by SarikaSinha** [comment]
Help and N&N to e revisited after all the polishing is done.
Help needs to be moved to platform, currently it is in jdt.
https://github.com/eclipse-platform/eclipse.platform.common/tree/master/bundles/org.eclipse.platform.doc.isv/guide

**Comment by mihnita** [comment]
Or does it mean "OK to submit this", BUT move things "after all the polishing is done"
(so after this PR is in, move the whole Run/Debug "tree")

**Comment by mihnita** [comment]
OK, I didn't see the drag & drop notice either :-)

**Comment by SarikaSinha** [comment]
My mistake, It will remain in JDT itself. Ideally it should have been under Platform but I don't see a point in changing that now.
We need to update the current PR after the polishing is done so that we have the final Help documentation.

**Comment by mihnita** [comment]
From my side I think I've addressed everything.

Then resolved the request from change
(although I think that the practice there is team dependent.
For some project you say "Done" when implemented, and the one who opened the request resolves it when they are satisfied with the fix)
I don't know what is the  custom here. 

Anyway...
From my side this is done. GitHub still says "Changes requested", but I can't find anything unresolved.

And this is also read, in my opinion:
https://github.com/eclipse-platform/eclipse.platform.debug/pull/62

**Comment by mihnita** [comment]
@iloveeclipse, @SarikaSinha 

Gentle ping.

I think I did all the changes, and marked them as resolved.
But for some reason GitHub still says "Changes requested".
I checked everything, but I don't fine any change not marked resolved.

Can you please take a look?

Thank you,
M.

**Comment by iloveeclipse** [comment]
Sorry, I'm pretty busy, we have troubles with https://github.com/eclipse-platform/eclipse.platform/issues/77.

**Comment by mihnita** [comment]
No problem. 
Thank you both!

**Pull request merged**